### PR TITLE
Fix positioning error in scanByPrefBack (SYN-7463)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

**Review by vEpiphyte — APPROVED** [review]

File: synapse/lib/lmdbslab.py [hunk 1]
```diff
@@ -1367,6 +1367,10 @@ def scanByPrefBack(self, byts, db=None):
                 if not scan.set_range(nextbyts):
                     return
 
+                if scan.atitem[0] == nextbyts:
```
Comment (mikemoritz) [inline synapse/lib/lmdbslab.py:1370]: minor nit, but couldn't this if statement be combined since there's nothing to do if scan.next_key()?

Comment (Cisphyx) [reply synapse/lib/lmdbslab.py:1370]: No, this is checking if `set_range` landed on exactly `nextbyts` and moving to the next key if it did (and if that failed returning).

Comment (Cisphyx) [reply synapse/lib/lmdbslab.py:1370]: If we didn't land on exactly `nextbyts` we're already on the key we want

Comment (mikemoritz) [reply synapse/lib/lmdbslab.py:1370]: whoops, didn't read ahead far enough :)

**Review by mikemoritz — APPROVED** [review]

**Pull request merged**